src/modules/_nomodule/PixelManipulation.js

```diff
@@ -314,8 +314,10 @@ module.exports = function PixelManipulation(image, options) {
       else resolvedFrames++;
 
       if (options.extraManipulation){
-        frames[f] = options.extraManipulation(framePix, setRenderState, generateOutput) || framePix; // extraManipulation is used to manipulate each pixel individually.
-        perFrameShape = frames[f].shape;
+        getDataUri(frames[f], options.format).then(datauri => {
```

Comment:
What is the data URI used for?

Reply:
in the module (fisheye) we need an image source for distorter to set the image. One option was to put pixels into canvas and get the url using canvas.todataurl() but this failed for gifs as it was making the transpose of that array. Hence to counter that i directly used the source url without using canvas method